config-model/src/main/java/com/yahoo/vespa/model/search/IndexedSearchCluster.java

```diff
@@ -216,59 +214,42 @@ public String toString() {
 
     /**
      * Class used to retrieve combined configuration from multiple document databases.
-     * It is not a direct {@link com.yahoo.config.ConfigInstance.Producer} of those configs,
+     * It is not a direct {@link ConfigInstance.Producer} of those configs,
      * that is handled (by delegating to this) by the {@link IndexedSearchCluster}
      * which is the parent to this. This avoids building the config multiple times.
      */
-    public static class MultipleDocumentDatabasesConfigProducer
-            extends TreeConfigProducer<MultipleDocumentDatabasesConfigProducer>
-            implements AttributesConfig.Producer,
-                       IndexInfoConfig.Producer,
-                       IlscriptsConfig.Producer,
-                       SchemaInfoConfig.Producer,
-                       RankProfilesConfig.Producer {
-        private final List<DocumentDatabase> docDbs;
-
-        private MultipleDocumentDatabasesConfigProducer(TreeConfigProducer<?> parent, List<DocumentDatabase> docDbs) {
-            super(parent, "union");
-            this.docDbs = docDbs;
-        }
+    private record Join(List<DocumentDatabase> docDbs) {
 
-        @Override
         public void getConfig(IndexInfoConfig.Builder builder) {
-            for (DocumentDatabase docDb : docDbs) {
-                docDb.getConfig(builder);
+                for (DocumentDatabase docDb : docDbs) {
+                    docDb.getConfig(builder);
+                }
             }
-        }
 
-        @Override
-        public void getConfig(SchemaInfoConfig.Builder builder) {
-            for (DocumentDatabase docDb : docDbs) {
-                docDb.getConfig(builder);
+            public void getConfig(SchemaInfoConfig.Builder builder) {
```

Comment:
Something wrong with the indentation? Applies for the functions below as well.

Reply:
Good point, I wonder if this happened the last time intellij converted to a record too.